linkerd/app/src/dst/resolve.rs

```diff
@@ -1,22 +1,24 @@
-use indexmap::IndexSet;
+use ipnet::{Contains, IpNet};
 use linkerd2_app_core::{
     dns::Suffix,
     dst::DstAddr,
     exp_backoff::{ExponentialBackoff, ExponentialBackoffStream},
     proxy::{api_resolve as api, resolve::recover},
-    request_filter, Error, Recover,
+    request_filter, Addr, Error, Recover,
 };
+use std::net::IpAddr;
 use std::sync::Arc;
 use tower_grpc::{generic::client::GrpcService, Body, BoxBody, Code, Status};
 
 pub type Resolve<S> = request_filter::Service<
-    PermitNamesInSuffixes,
+    PermitConfiguredDsts,
     recover::Resolve<BackoffUnlessInvalidArgument, api::Resolve<S>>,
 >;
 
 pub fn new<S>(
```

Comment:
Nit/TIOLI: this is kind of a lot of arguments...I know this method isn't called very often, but the callsite might be easier to read if there was a builder, or at least a struct with a bunch of named fields? maybe it's not worth it...